runner/runner/main.py

```diff
@@ -181,67 +185,129 @@ def load_tests(config: Mapping[str, Any]) -> None:
+        ## Pathify 'src_path' for compilation
+        openxr_app_path     = pathify(openxr_app_obj["app"]["src_path"], root_dir, cache_path, True , True)
+        openxr_app_bin_path = openxr_app_path / openxr_app_obj["app"]["bin_subpath"]
+    else:
+        ## Get the full path to the 'app' binary
+        openxr_app_path     = None
+        openxr_app_bin_path = pathify(openxr_app_obj["app"], root_dir, cache_path, True, True)
+
+    ## Compile the OpenXR app if we received an 'app' with 'src_path'
+    if openxr_app_path:
+        cmake(
+            openxr_app_path,
+            openxr_app_path / "build",
+            dict(CMAKE_BUILD_TYPE=cmake_profile, **openxr_app_config),
+        )
+
+    if not openxr_app_bin_path.exists():
+        raise RuntimeError(f"{action_name} Failed to build openxr_app (mainline={is_mainline}, path={openxr_app_bin_path})")
+
+    if is_mainline:
+        monado_target_name : str  = "monado-service"
+        monado_target_dir  : Path = monado_path / "build" / "src" / "xrt" / "targets" / "service"
+        monado_target_path : Path = monado_target_dir / monado_target_name
+
+        if not monado_target_path.exists():
+            raise RuntimeError(f"[{action_name}] Failed to build monado (mainline={is_mainline}, path={monado_target_path})")
+
+        env_popen: Mapping[str, str] = dict(**os.environ, **env_monado)
+
+        ## Open the Monado service application in the background
+        monado_socket = Path("/run/user/1000/monado_comp_ipc")
+        if monado_socket.exists():
+            ## The Monado service will crash if a previous instance is left behind
+            monado_socket.unlink()
+        monado_service_proc = subprocess.Popen([str(monado_target_path)], env=env_popen, cwd=Path())
+
     subprocess_run(
-        [str(openxr_app_path / "build" / "./openxr-example")],
+        [str(openxr_app_bin_path)],
         env_override=dict(
-            XR_RUNTIME_JSON=str(monado_path / "build" / "openxr_monado-dev.json"),
-            ILLIXR_PATH=str(runtime_path / f"plugin.{profile}.so"),
-            ILLIXR_COMP=":".join(map(str, plugin_paths)),
-            ILLIXR_DATA=str(data_path),
             ILLIXR_DEMO_DATA=str(demo_data_path),
             ILLIXR_OFFLOAD_ENABLE=str(enable_offload_flag),
             ILLIXR_ALIGNMENT_ENABLE=str(enable_alignment_flag),
             ILLIXR_ENABLE_VERBOSE_ERRORS=str(config["enable_verbose_errors"]),
             ILLIXR_ENABLE_PRE_SLEEP=str(config["enable_pre_sleep"]),
             KIMERA_ROOT=config["action"]["kimera_path"],
             REALSENSE_CAM=str(realsense_cam_string),
+            **env_monado,
         ),
         check=True,
     )
 
+    if is_mainline:
+        ## Close and clean up the Monado service application
+        try:
+            outs, errs = monado_service_proc.communicate(timeout=1)
+        except subprocess.TimeoutExpired:
+            monado_service_proc.kill()
+            outs, errs = monado_service_proc.communicate()
```

Comment:
`monado-service` cleanup

Reply:
Let's write `out` and `errs` to a `sys.stdout` and `sys.stderr`.

Reply:
Is it always doing the `except` branch?

Maybe we should send a `SIGTERM` first, like `monado_service_proc.terminate()`.

This might also help with the zombie socket.

Reply:
This is now fixed.